NetworkPkg/Ip6Dxe/Ip6Nd.c

```diff
@@ -1281,15 +1302,29 @@ Ip6SendNeighborAdvertise (
   }
 
   Target = (EFI_IPv6_ADDRESS *)NetbufAllocSpace (Packet, sizeof (EFI_IPv6_ADDRESS), FALSE);
-  ASSERT (Target != NULL);
+  // MU_CHANGE [BEGIN] - CodeQL change
+  if (Target == NULL) {
+    ASSERT (Target != NULL);
+    NetbufFree (Packet);
+    return EFI_OUT_OF_RESOURCES;
+  }
+
+  // MU_CHANGE [END] - CodeQL change
   IP6_COPY_ADDRESS (Target, TargetIp6Address);
 
   LinkLayerOption = (IP6_ETHER_ADDR_OPTION *)NetbufAllocSpace (
                                                Packet,
                                                sizeof (IP6_ETHER_ADDR_OPTION),
                                                FALSE
                                                );
-  ASSERT (LinkLayerOption != NULL);
+  // MU_CHANGE [BEGIN] - CodeQL change
+  if (LinkLayerOption == NULL) {

```

Comment:
Also add `NetBufFree (Target);` here.

Reply:
These `NetbufAllocSpace()` calls appear to be carving out space from an already allocated buffer so I'll drop my comment on them.